ONSdigital/dp-filter-api
# Feature/add assert filter type middleware

### What

Added assert.FilterType middleware to intercept requests on flexible type filters to be fowarded to `dp-cantabular-filter-flex-api`

Only added to GET /filter/id path for now, as the equivalent endpoints are developed in `dp-cantabular-filter-flex-api` we'll need to add this middleware to those routes too

### How to review

Check changes make sense, tests pass

To test manually:
- Bring up Cantabular Import journey and checkout to this branch
- make POST request to /filters with a cantabular_flexible_filter type dataset to create filter
- Test request is forwarded to dp-filter-flexapi when making GET request to /filters/{id} (will throw an error if not, also check the logs for the `dp-cantabular-filter-flex-api` service to be sure

### Who can review

Describe who worked on the changes, so that other people can review.
